File: personal/23_02_01/2798/2798.py
# 블랙잭
# 3명이서 게임
# N장의 카드를 숫자가 보이도록 바닥에 놓음 > 후에 딜러는 숫자 M을 외침


import sys

input = sys.stdin.readline

n, m = map(int, input().split())

# cards = []
arr = list(map(int, input().split()))

# print(arr)
result = 0
for i in range(n):
    for j in range(i+1, n):
        for k in range(j+1, n):
            sum = arr[i] + arr[j] + arr[k]
            if sum > m:
                continue
            else:
                result = max(result, sum)

print(result)

File: JS코테준비/B/2798(블랙잭)/2798.js
const { log } = require("console");
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let input = fs.readFileSync(filePath).toString().split("\n");
let n = input[0].split(" ").map((item) => +item)[0];
let m = input[0].split(" ").map((item) => +item)[1];
let cards = input[1].split(" ").map(Number);
// N(카드 갯수) M(이 수를 넘지 않아야 함)
// 정수가 적혀 있는 N장의 카드 5개 나열
let max = 0;

loop1: for (let i = 0; i < n - 2; i++) {
  for (let j = i + 1; j < n - 1; j++) {
    for (let k = j + 1; k < n; k++) {
      const sum = cards[i] + cards[j] + cards[k];
      if (sum > max && sum <= m) {
        max = sum;
      }
      if (max === m) {
        break loop1;
      }
    }
  }
}
console.log(max);
